module/shardingManager.ts

```diff
@@ -563,7 +563,10 @@ export async function handleDiscordPayload(
       }
 
       if (data.t === "PRESENCE_UPDATE") {
-        return eventHandlers.presenceUpdate?.(data.d as PresenceUpdatePayload);
+        const payload = data.d as PresenceUpdatePayload;
+        const oldPresence = cache.presences.get(payload.user.id) ?? null;
```

Comment:
```suggestion
        const oldPresence = cache.presences.get(payload.user.id);
```
Forcing a `null` here is unnecessary and TS can handle undefined just as nicely and it'll make typings easier as well. Also as consistency measure the rest of the lib avoids null whenever possible as well for `undefined` except the in the cases where `null` is necessary.